test/pylib/s3_proxy.py

```diff
@@ -154,6 +165,7 @@ def process_request(self):
 
             if policy.error_count >= policy.max_errors:
                 policy.should_fail = False
+                policy.server_should_fail = False
```

Comment:
I don't understand this change - where do we ever set server_should_fail to True? Why would we do that? How is it related to the commit message which is something about Seastar's **client** behavior?

Reply:
Here https://github.com/scylladb/scylladb/pull/21933/files#diff-b5f846f1d017bc6def9ecb3351c41a9124c29a002efc4f5c813b9c5bbed42fb2R124

It is intended to emulate "connection reset by peer" error which was not handled correctly